### PR TITLE
build: update to Micronaut 4

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

**Comment by sdelamo** [comment]
@timyates could you look into this?

**Comment by timyates** [comment]
Still failing, I suspect it's an issue in core

**Comment by timyates** [comment]
@sdelamo Fixed.

I had to pass the ConversionService in to the RedisSessionStore, which meant I had to change the public constructor

After a discussion in the meeting, there was no preference to simply making the change over deprecating the old constructor, so I went with the former and added it to the accepted-api-changes

File: gradle.properties [hunk 1]
```diff
@@ -2,11 +2,11 @@ projectVersion=6.0.0-SNAPSHOT
 projectGroup=io.micronaut.redis
 
 micronautDocsVersion=2.0.0
-micronautVersion=3.7.4
-micronautTestVersion=3.7.1
+micronautVersion=4.0.0-SNAPSHOT
```
Comment (wetted) [inline gradle.properties:5]: I started moving all these version properties to libs.versions.toml, e.g.
```
micronaut-docs = "2.0.0"
micronaut = "4.0.0-SNAPSHOT"
micronaut-test = "4.0.0-SNAPSHOT"
groovy = "4.0.6"
spock = "2.2-groovy-4.0"
```

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
-micronaut-inject-java = { module = 'io.micronaut:micronaut-inject-java' }
-micronaut-management = { module = 'io.micronaut:micronaut-management' }
-micronaut-micrometer = { module = 'io.micronaut.micrometer:micronaut-micrometer-core'}
-micronaut-session = { module = 'io.micronaut:micronaut-session' }
+micronaut-cache = { module = "io.micronaut.cache:micronaut-cache-bom", version.ref = "micronaut-cache" }
+micronaut-micrometer = { module = "io.micronaut.micrometer:micronaut-micrometer-bom", version.ref = "micronaut-micrometer" }
+micronaut-serde = { module = "io.micronaut.serde:micronaut-serde-bom", version.ref = "micronaut-serde" }
+micronaut-session = { module = "io.micronaut.session:micronaut-session-bom", version.ref = "micronaut-session" }
 
 projectreactor = { module = 'io.projectreactor:reactor-core' }
```
Comment (wetted) [inline gradle/libs.versions.toml:21]: I think this can be eliminated in favor of `mn.reactor`

File: redis-lettuce/build.gradle [hunk 3]
```diff
@@ -1,34 +1,36 @@
...
+    compileOnly mnMicrometer.micronaut.micrometer.core
     compileOnly libs.redis.embedded
 
-    testAnnotationProcessor libs.micronaut.inject.java
+    testAnnotationProcessor mn.micronaut.inject.java
 
     testImplementation platform(libs.testcontainers.bom)
     testImplementation libs.testcontainers.spock
 
     testImplementation libs.projectreactor
```
Comment (wetted) [inline redis-lettuce/build.gradle:21]: I think this can now be `mn.reactor`

**Review by wetted — CHANGES_REQUESTED** [review]

**Review by wetted — APPROVED** [review]

**Pull request merged**